lib/utils/background_push.dart

```diff
@@ -75,25 +75,28 @@ class BackgroundPush {
     onRoomSync ??= client.onSync.stream
         .where((s) => s.hasRoomUpdate)
         .listen((s) => _onClearingPush(getFromServer: false));
-    fcmSharedIsolate?.setListeners(onMessage: (message) {
-      Logs().d('BackgroundPush::onMessage(): $message');
-      final notification = _parseMessagePayload(message);
-      pushHelper(
-        notification,
-        client: client,
-        l10n: l10n,
-        activeRoomId: router?.currentState?.pathParameters['roomid'],
-        onSelectNotification: goToRoom,
-      );
-      Logs().d('BackgroundPush::onMessage(): finished pushHelper');
-    });
     if (Platform.isAndroid) {
       UnifiedPush.initialize(
         onNewEndpoint: _newUpEndpoint,
         onRegistrationFailed: _upUnregistered,
         onUnregistered: _upUnregistered,
         onMessage: _onUpMessage,
       );
+      fcmSharedIsolate?.setListeners(onMessage: (message) {
+        onReceiveNotification(message);
+      });
+    } else if (Platform.isIOS) {
+      apnChannel.setMethodCallHandler((call) async {
+        Logs().v('[Push] Received APN call: $call');
+        if (call.method == 'willPresent') {
+          onReceiveNotification(call.arguments);
+        } else if (call.method == 'didReceive') {
+          onReceiveNotification(call.arguments);
+          // Go to room because user selected the notification
+          goToRoom(call.arguments['room_id']);
+        }
+      });
+      getIosInitialNoti();
```

Comment:
Why did you put it here? Why not at the top?
```
else if (Platform.isIOS) {
  getIosInitialNoti();
  apnChannel.setMethodCallHandler...
}
```

Reply:
no worries, both func just initial, one for listeners and one for initial noti
Android do same thing, no impact